args/audio/components/audio_source.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+#include <AL/al.h>
+#include <AL/alc.h>
+#include <audio/data/audio_segment.hpp>
+
+namespace args::audio
+{
+    struct audio_source
+    {
+        friend class AudioSystem;
+    public:
+        enum sound_properties
+        {
+            pitch = 1,
+            gain = 2,
+        };
```

Comment:
if this is used more than once in the audio library, maybe make it a global enum in the args::audio namespace instead of a member of the audio_source class.